### DISCUSSION TITLE
Sparkle In a Shared Framework

### DISCUSSION THREAD

**Discussion by bdkjones** [sparkle-project/Sparkle#2339] [Q&A]
## Scenario:

I use a custom Sparkle driver because my "update available" window needs to inform a user if their license covers the proposed update (my apps' licenses include updates for a period of time, then must be renewed for more updates).

I have many apps that work this way, so I would like to build a reusable Framework that includes all of my Sparkle customizations so I don't have to redo this work  for each app.

## Question:

Is this possible using Sparkle 2.x? I want to update the "host" bundle--the main application that imports my reusable custom Framework. Sparkle would be in the imported Framework, but not the host app.

I've read the docs related to updating a plug-in or preference pane bundle, here: https://sparkle-project.org/documentation/bundles/ but I'm unsure this matches my situation.

The apps are not sandboxed.

**Comment by zorgiepoo** [comment]
Is your custom framework that embeds Sparkle laid down somewhere on the system for multiple apps to access, or is the custom framework embedded inside each one of your apps (the latter being more likened to macOS style of apps desiring to be more self contained). Either way I don't think this is much different than using Sparkle normally so I can't think of issues; it just changes where you link against Sparkle. Of course you should test/try it out. Things get more complex when you have multiple processes or plug-ins, which doesn't seem like this case.

**Comment by bdkjones** [comment]
@zorgiepoo My plan was to embed the framework into each app. It's just a standard Framework (via the Xcode "New Framework" template). The usual collection of various classes, nibs, and some SwiftUI. I don't like sharing libraries across apps because that becomes a nightmare when the library needs changes.

I'm migrating all my customizations to a Framework now to try it out, so I'll report back on how it goes. Thanks!
